openedx/core/djangoapps/discussions/views.py

```diff
@@ -0,0 +1,57 @@
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys import InvalidKeyError
+from rest_framework import serializers
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from openedx.core.lib.api.permissions import IsStaff
+
+from .models import DiscussionsConfiguration
+from .models import ProviderFilter
+
+
+class ConfigurationView(APIView):
+    authentication_classes = (JwtAuthentication, SessionAuthenticationAllowInactiveUser)
+    permission_classes = (IsStaff,)
+
+    class Serializer(serializers.BaseSerializer):
+        def to_representation(self, data):
+            active = ''
+            if data['configuration'].enabled:
+                active = data['configuration'].provider_type
+            return {
+                'providers': {
```

Comment:
So this is the shape that's in the frontend, which obviously has a bunch of test data fleshing it out:

```
{
  features: [
    {
      id: 'lti',
      name: 'LTI Integration',
    },
    {
      id: 'discussion-page',
      name: 'Discussion Page',
    },
    {
      id: 'embedded-course-sections',
      name: 'Embedded Course Sections',
    },
    {
      id: 'embedded-course-units',
      name: 'Embedded Course Units',
    },
    {
      id: 'wcag-2.1',
      name: 'WCAG 2.1 Support',
    },
  ],
  apps: [
    {
      id: 'piazza',
      name: 'Piazza',
      logo: 'https://piazza.com/images/splash2/topbar/piazza_logo_blue.png',
      description: 'Piazza is designed to connect students, TAs, and professors so every student can get the help they need when they need it.',
      support_level: 'Partial support',  # I think we'll actually compute this in the frontend
      is_available: true, # Used as a teaser for future integrations, maybe not necessary for now.
      documentation_url: 'https://localhost/docs',
      feature_ids: [
        'lti',
        'discussion-page',
        'embedded-course-sections',
        'wcag-2.1',
      ],
    }
    // Other apps go here, same schema as Piazza above
  ],
  active_app_id: 'piazza',
}
```

`features` tells us the master list of capabilities a discussions app can have, and the `feature_ids` list in an app tells. us what it supports.

I called them `apps` throughout the frontend, but maybe I should back off that wording... I could be convinced either way.  It seems like a friendlier/shorter term than providers. ¯\_(ツ)_/¯

Reply:
In terms of internationalization, we could also just pass the feature IDs to the frontend and it could encode a set of display strings for them/localize them.  It's not perfect, but assuming the list doesn't change too often, it's good enough.